audisp-tacplus.c

```diff
@@ -605,10 +489,7 @@ static void get_acct_record(auparse_state_t *au, int type)
      * loguser is always set, we bail if not.  For ANOM_ABEND, tty may be
      *  unknown, and in some cases, host may be not be set.
      */
-    send_tacacs_acct(loguser, tty?tty:"UNK", host?host:"UNK", logbase, acct_type, taskno);
```

Comment:
>host?host:"UNK" [](http://example.com/codeflow?start=45&length=15)

You `host` is same as original `host`, right? Are you fixing a bug?

Reply:
The code change is caused by how we get 'host'.
The upstream project using following code to get host from libtacplus_map:
loguser = lookup_logname(NULL, auid, session, &host, NULL);

Because SONiC not provide libtacplus_map, so we rewrite the logic to get host, and set host value to 'UNK' if get host failed:
    /* get hostname. */
    char host[HOST_NAME_MAX];
    memset(host, 0, sizeof(host));
    if (gethostname(host, sizeof(host)) != 0)
    {
        strncpy(host, unknown_hostname, sizeof(host)-1);
    }

Reply:
My concern is that you don't need to copy `unknown_hostname` string. The original `host?host:"UNK"` works well with your new flow.

Reply:
In the new code, host always point to a buffer on stack, so the 'host?host:"UNK"' will not work anymore.
I will change the lookup_logname to assign hostname when we can find hostname. and keep the upstream code here.